jax_galsim/gsobject.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+import jax.numpy as jnp
+
+from jax._src.numpy.util import _wraps
+import galsim as _galsim
+
+from jax_galsim.gsparams import GSParams
+
+
+@_wraps(_galsim.GSObject)
+class GSObject:
+    def __init__(self, *, gsparams=None, **params):
+        self._params = params  # Dictionary containing all traced parameters
+        self._gsparams = gsparams  # Non-traced static parameters
+
+    @property
+    def flux(self):
+        """The flux of the profile."""
+        return self._params["flux"]
+
+    @property
+    def gsparams(self):
+        """A `GSParams` object that sets various parameters relevant for speed/accuracy trade-offs."""
+        return self._gsparams
+
+    @property
+    def params(self):
+        """A Dictionary object containing all parameters of the internal represention of this object."""
+        return self._params
+
+    @property
+    def maxk(self):
+        """The value of k beyond which aliasing can be neglected."""
+        return self._maxk
+
+    @property
+    def stepk(self):
+        """The sampling in k space necessary to avoid folding of image in x space."""
+        return self._stepk
+
+    @property
+    def nyquist_scale(self):
+        """The pixel spacing that does not alias maxk."""
+        return jnp.pi / self.maxk
+
+    @property
+    def has_hard_edges(self):
+        """Whether there are any hard edges in the profile, which would require very small k
+        spacing when working in the Fourier domain.
+        """
+        return self._has_hard_edges
+
+    @property
+    def is_axisymmetric(self):
+        """Whether the profile is axially symmetric; affects efficiency of evaluation."""
+        return self._is_axisymmetric
+
+    @property
+    def is_analytic_x(self):
+        """Whether the real-space values can be determined immediately at any position without
+        requiring a Discrete Fourier Transform.
+        """
+        return self._is_analytic_x
+
+    @property
+    def is_analytic_k(self):
+        """Whether the k-space values can be determined immediately at any position without
+        requiring a Discrete Fourier Transform.
+        """
+        return self._is_analytic_k
+
+    def __add__(self, other):
+        """Add two GSObjects.
+
+        Equivalent to Add(self, other)
+        """
+        from jax_galsim.sum import Sum
+
+        return Sum([self, other])
+
+    def __eq__(self, other):
+        return self is other or (
+            isinstance(other, self.__class__)
```

Comment:
yes you are correct actually, isinstance will be true for a derived type, which is not what we want here. We want this to be true only if the 2 objects are exactly the same type and with the same flattened representation.

Reply:
ok, I fixed it by checking type(other) instead, i think this works.

This being said.... this made me think a bit more... although this implementation will work, it will not catch problems with the tree flatening  / unflatening mechanism. if for instance we forget to add all revelant params to the pytree, it might test equal, but the object may be different.

I'm wondering if we shouldn't keep the old galsim mechanism instead:
https://github.com/GalSim-developers/GalSim/blob/ece3bd32c1ae6ed771f2b489c5ab1b25729e0ea4/galsim/gaussian.py#L116 

which implies redefining this test specifically for each obecjt, so taht we can for inistance test an objct is still thee same after jitting.... 

I have mixed feelings because it's also a good opporrtuniniy to remove a good deal of redundant code....

Reply:
>  it will not catch problems with the tree flatening / unflatening mechanism. if for instance we forget to add all revelant params to the pytree, it might test equal, but the object may be different.

Sorry not sure I completely follow. You mean for new GSObjects we define? If we forget the add the relevant params to the pytree then this would be a major bug in our implementation of that object right? So hopefully other tests catch these? 

> which implies redefining this test specifically for each obecjt, so taht we can for inistance test an objct is still thee same after jitting....

Sorry also confused by this,  you can't jit an object by itself right? You can only jit functions of objects...

Reply:
> I have mixed feelings because it's also a good opporrtuniniy to remove a good deal of redundant code....

I feel like it would be possible to avoid but then we might have to have more comprehensive tests for equality